### ISSUE TITLE
Nonce in XChaCha20 Poly1305 truncated after calling new()

### ISSUE THREAD

**Issue by lhhaber** [Legrandin/pycryptodome#722]
Howdy,

When providing a 24 byte nonce to `ChaCha20_Poly1305`, retrieving the nonce via `cipher.nonce` truncates it to 12 bytes.

Example of issue:
```
>>> from Crypto.Cipher import ChaCha20_Poly1305
>>> cipher = ChaCha20_Poly1305.new(key=b"a"*32, nonce=b"a"*24)
>>> cipher.nonce
b'\x00\x00\x00\x00aaaaaaaa'
>>> len(cipher.nonce)
12
```

Relevant documentation: https://pycryptodome.readthedocs.io/en/latest/src/cipher/chacha20_poly1305.html
Similar to prior resolved issue: https://github.com/Legrandin/pycryptodome/issues/500#issue-802554974

**Comment by Legrandin** [comment]
Thanks, fixed with b6fe8d11ba05e2019789919cc56bd9e6887987e3

**Comment by Legrandin** [comment]
Fixed in v3.18.0
